package.json

```diff
@@ -26,7 +26,8 @@
     "lint": "eslint 'src/**/*.ts' 'tests/**/*.ts'",
     "test": "pnpm jest",
     "unit-test": "pnpm jest tests/unit",
-    "e2e-test": "pnpm jest tests/e2e",
+    "e2e-test": "pnpm run _e2e-test:publish-contracts && jest tests/e2e",
+    "_e2e-test:publish-contracts": "ts-node -r dotenv/config ./scripts/publishANSContracts.mts",
```

Comment:
Pulling and building the contract isn't a cheap operation. Just wanted to highlight that for consideration.

Reply:
note: we also need to make sure we run it on CI https://github.com/aptos-labs/aptos-core/blob/main/.github/actions/run-ts-sdk-e2e-tests/action.yaml#L38